### PR TITLE
TT smart save

### PR DESCRIPTION
**Pull request by mstembera**
STC 2MB
LLR: 2.96 (-2.94,2.94) [-1.50,4.50]
Total: 21132 W: 4108 L: 3946 D: 13078
http://tests.stockfishchess.org/tests/view/5547d59f0ebc5940ca5d6883

LTC 8MB
LLR: 2.97 (-2.94,2.94) [0.00,6.00]
Total: 13381 W: 2149 L: 1987 D: 9245
http://tests.stockfishchess.org/tests/view/5549b5a80ebc5940ca5d68b9

STC 16MB regression w/ zero effective hash pressure
LLR: 2.96 (-2.94,2.94) [-5.00,0.00]
Total: 18944 W: 3607 L: 3564 D: 11773
http://tests.stockfishchess.org/tests/view/554b0fda0ebc5940ca5d68ea

### CODE REVIEW

**Comment by joergoster** [comment]
Congrats!

**Comment by glinscott** [comment]
Very nice, congratulations!  Looks good :).

File: src/tt.h [hunk 1]
```diff
@@ -43,14 +43,29 @@ struct TTEntry {
...
+          key16     = (uint16_t)(k >> 48);
+          move16    = (uint16_t)m;
+          value16   = (int16_t)v;
+          eval16    = (int16_t)ev;
+          genBound8 = (uint8_t)(g | b);
+          depth8    = (int8_t)d;
+      }
+      else
+      {   
+          if (m) move16 = (uint16_t)m; // Preserve any existing move for the same position
```
Comment (lucasart) [inline src/tt.h]: Can you use the use the standard indentation, please ?

    if (m) // Preserve any existing move for the same position
        move16 = (uint16_t)m;

File: src/tt.h [hunk 2]
```diff
@@ -43,14 +43,29 @@ struct TTEntry {
...
+
+          // Don't overwrite more valuable values
+          if (d+2 > depth8 || g != (genBound8 & 0xFC) || b == BOUND_EXACT)
+          {
+              value16   = (int16_t)v;
+              genBound8 = (uint8_t)(g | b);
+              depth8    = (int8_t)d;
+          }
+
+          if (ev != VALUE_NONE) eval16 = (int16_t)ev;
```
Comment (lucasart) [inline src/tt.h]: same here:

    if (ev != VALUE_NONE)
        eval16 = (int16_t)ev;

**Comment by zamar** [comment]
approved.

**Pull request closed without merging**